nodes/list/set_list_element.py

```diff
@@ -43,22 +43,29 @@ def drawAdvanced(self, layout):
 
     def getExecutionCode(self):
         yield "self.errorMessage = ''"
+        if self.isTuple: yield "_list = list(_list)"
+        
         if self.allowNegativeIndex:
             if self.clampIndex:
-                yield "if len(list) != 0: list[min(max(index, -len(list)), len(list) - 1)] = element"
+                yield "if len(_list) != 0: _list[min(max(index, -len(_list)), len(_list) - 1)] = element"
             else:
-                yield "if -len(list) <= index <= len(list) - 1: list[index] = element"
+                yield "if -len(_list) <= index <= len(_list) - 1: _list[index] = element"
         else:
             if self.clampIndex:
-                yield "if len(list) != 0: list[min(max(index, 0), len(list) - 1)] = element"
+                yield "if len(_list) != 0: _list[min(max(index, 0), len(_list) - 1)] = element"
             else:
-                yield "if 0 <= index <= len(list) - 1: list[index] = element"
+                yield "if 0 <= index <= len(_list) - 1: _list[index] = element"
         yield "else: self.errorMessage = 'Index out of range'"
 
     def edit(self):
         dataType = self.getWantedDataType()
         self.assignType(dataType)
 
+    def isTuple(self):
+        listInput = self.inputs["List"].dataOrigin
+        if listInput is not None and isLimitedList(listInput.bl_idname): return True
```

Comment:
this line throws an exception when listInput is None, right?
If you would do the same thing in C code it would actually work, but Python always evaluates everything in the conditional expression.
listInput.bl_idname   throws the error here